include/wsrep/provider.hpp

```diff
@@ -300,7 +327,8 @@ namespace wsrep
         virtual enum status commit_order_enter(const wsrep::ws_handle&,
                                                const wsrep::ws_meta&) = 0;
         virtual int commit_order_leave(const wsrep::ws_handle&,
-                                       const wsrep::ws_meta&) = 0;
+                                       const wsrep::ws_meta&,
+                                       const wsrep::mutable_buffer& err) = 0;
```

Comment:
Wouldn't it make sense to pass `err` as `const_buffer`?

Reply:
This is a question of API purity, I guess. My decision here was that if we already have to have a `mutable_buffer`, why bother with `const_buffer`, especially given that this code path is executed for local transactions as well?

Reply:
`leave_toi()` below uses `const_buffer`, it would be nice to have consistent way to pass error buffer. Maybe `mutable_buffer` would be more appropriate then.

Though, I was thinking to have separate `error_buffer` class with specialized interface, just for type safety. Not sure if it is an overkill or not.

Reply:
`leave_toi()` fixed